src/hotspot/cpu/x86/sharedRuntime_x86_64.cpp

```diff
@@ -1259,99 +1259,129 @@ static void verify_oop_args(MacroAssembler* masm,
 
 // defined in stubGenerator_x86_64.cpp
 OopMap* continuation_enter_setup(MacroAssembler* masm, int& stack_slots);
-void fill_continuation_entry(MacroAssembler* masm);
+void fill_continuation_entry(MacroAssembler* masm, Register reg_cont_obj, Register reg_flags);
 void continuation_enter_cleanup(MacroAssembler* masm);
 
-// enterSpecial(Continuation c, boolean isContinue, boolean isVirtualThread)
-// On entry: c_rarg1 -- the continuation object
-//           c_rarg2 -- isContinue
-//           c_rarg3 -- isVirtualThread
+static void check_continuation_enter_argument(VMReg actual_vmreg,
+                                              Register expected_reg,
+                                              const char* name) {
+  assert(!actual_vmreg->is_stack(), "%s cannot be on stack", name);
+  assert(actual_vmreg->as_Register() == expected_reg,
+         "%s is in unexpected register: %s instead of %s",
+         name, actual_vmreg->as_Register()->name(), expected_reg->name());
+}
+
 static void gen_continuation_enter(MacroAssembler* masm,
                                  const methodHandle& method,
                                  const BasicType* sig_bt,
```

Comment:
Should we remove these two parameters? They were used on a commented out line which is now removed.